src/main/kotlin/com/terraformation/backend/accelerator/model/SustainableDevelopmentGoal.kt

```diff
@@ -0,0 +1,42 @@
+package com.terraformation.backend.accelerator.model
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonValue
+
+val startingDigitRegex = Regex("^\\d+")
+
+enum class SustainableDevelopmentGoal(val displayName: String) {
+  NoPoverty("1. No Poverty"),
+  ZeroHunger("2. Zero Hunger"),
+  GoodHealth("3. Good Health and Well-Being"),
+  QualityEducation("4. Quality Education"),
+  GenderEquality("5. Gender Equality"),
+  CleanWater("6. Clean Water and Sanitation"),
+  AffordableEnergy("7. Affordable and Clean Energy"),
+  DecentWork("8. Decent Work and Economic Growth"),
+  Industry("9. Industry, Innovation, and Infrastructure"),
+  ReducedInequalities("10. Reduced Inequalities"),
+  SustainableCities("11. Sustainable Cities and Communities"),
+  ResponsibleConsumption("12. Responsible Consumption and Production"),
+  ClimateAction("13. Climate Action"),
+  LifeBelowWater("14. Life Below Water"),
+  LifeOnLand("15. Life on Land"),
+  Peace("16. Peace, Justice, and Strong Institutions"),
+  Partnerships("17. Partnerships for the Goals");
+
+  @get:JsonValue val sdgNumber: Int = startingDigitRegex.find(displayName)?.value?.toInt()!!
+
+  companion object {
+    private val bySdgNumber: Map<Int, SustainableDevelopmentGoal> by lazy {
+      SustainableDevelopmentGoal.entries.associateBy { it.sdgNumber }
+    }
+
+    @JsonCreator
+    @JvmStatic
+    fun forJsonValue(value: String): SustainableDevelopmentGoal {
```

Comment:
If you change `value`'s type to `Int`, that'd take care of the Graphite bot's comment as well as remove the need to use the regex. I believe `@JsonCreator` annotations work on methods with numeric primitive type arguments.

Reply:
`value` is actually coming from the variable text value in the database, so we either do the regex before calling this method, or do it in the method